### PR TITLE
Gamma armoury balance

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
The PR introduces some changes to gamma armoury:

- Readds buckshot speedloaders to shotgun cabinet
- Adds buckshot to (only) the gamma armoury autolathe
- Replaces ion rifles with multi-lens immolators
- Adds an extra set of glass and metal for the lathe
- Adds a second medbeam gun to the medbeam locker (don't cross them!)

## Why It's Good For The Game
The Gamma armoury is in a really weird state currently, us admins should only be calling it in major crises that require extra tools for the crew. These are usually biohazards or event threats.
With that in mind, the only actually good thing in the armoury is the mech (that is very strong in the right hands.)
Rubbershot for the combat shotguns is totally useless as very few major threats require it, let alone gamma level threats.
Ion **_rifles_**, not even carbines, are a total waste of space. Seriously, they help nobody and are not useful in any gamma level situation.

I've heard some say that gamma armoury should be tweaked by admins before sending, but if they will be doing that every time anyway because of these poor items within, it gets a bit silly.
It would be easier to make it stronger and encourage them to tweak it down in content if the situation is not as severe.

Currently, sending the gamma armoury is not very appealing.

## Changelog
:cl:
add: Added buckshot design to gamma armoury lathe
tweak: Replaced gamma armoury rubbershot speedloaders with buckshot speedloaders
tweak: Replaced ion rifles with multi-lens immolators
tweak: Changed amount of metal and glass for gamma armoury lathe
tweak: Changed number of medbeam guns given in gamma armoury from one to two
/:cl:

### CODE REVIEW

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Put the fixes and balance changes in different PRs.

PRs should be atomic.

**Comment by S34NW** [comment]
> Put the fixes and balance changes in different PRs.
> 
> PRs should be atomic.

Done, fixes incoming in another PR

**Comment by S34NW** [comment]
I made a big mess of that merge master, it's fixed now though

**Review by SteelSlayer — COMMENTED** [review]
**Edit: Turns out that the following issue is only an issue when you spawn the gamma autolathe directly, and not when it appears in the gamma armory. I have no idea why that's happening but my issue isn't technically an issue in this case.**

So, the gamma autolathe isn't actually receiving the buckshot design. I narrowed it down to this point of `adjust_hacked()`:

image

Before that loop, the `known_designs` list has the buckshot design, but it disappears after. I'm guessing it's because the buckshot doesn't have the `AUTOLATHE` `build_type`.

**Comment by S34NW** [comment]
> So, the gamma autolathe isn't actually receiving the buckshot design. I narrowed it down to this point of `adjust_hacked()`:
> 
> image
> 
> Before that loop, the `known_designs` list has the buckshot design, but it disappears after. I'm guessing it's because the buckshot doesn't have the `AUTOLATHE` `build_type`.

Are you sure? It works on local testing for me

**Comment by SteelSlayer** [comment]
You might want to include in the Description and/or CL that you added another stack of glass and metal, and another medibeam gun, just for the sake of documenting everything.

**Comment by S34NW** [comment]
> You might want to include in the Description and/or CL that you added another stack of glass and metal, and another medibeam gun, just for the sake of documenting everything.

Thanks, I'd actually forgotten this had been done.

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -78,8 +78,9 @@
 	RefreshParts()
 
 /obj/machinery/autolathe/upgraded/gamma/Initialize()
-    . = ..()
-    adjust_hacked(TRUE)
+	files = new /datum/research/autolathe/gamma(src)
+	. = ..()
+	adjust_hacked(TRUE)
```
Comment (AffectedArc07) [inline code/game/machinery/autolathe.dm:83]:
```suggestion
	. = ..()
	files = new /datum/research/autolathe/gamma(src)
	adjust_hacked(TRUE)
```

Keep init order sane.

Comment (S34NW) [reply code/game/machinery/autolathe.dm:83]: I'm trying to remember why this wasn't working... But it breaks stuff., I'll look at a more thorough workaround

**Review by AffectedArc07 — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Comment by S34NW** [comment]
oh shit

**Comment by S34NW** [comment]
oops

File: code/game/machinery/autolathe.dm [hunk 2]
```diff
@@ -78,8 +67,9 @@
 	RefreshParts()
 
 /obj/machinery/autolathe/upgraded/gamma/Initialize()
-    . = ..()
-    adjust_hacked(TRUE)
+	. = ..()
+	files = new /datum/research/autolathe/gamma(src)
```
Comment (dearmochi) [inline code/game/machinery/autolathe.dm:71]: Little bit out of scope but if a griefer wanted to, they could decon the autolathe and boom you can't craft buckshot anymore. Oh well.

**Review by dearmochi — APPROVED** [review]

**Pull request merged**